source/MaterialXGenShader/ShaderNode.cpp

```diff
@@ -381,7 +381,17 @@ void ShaderNode::initialize(const Node& node, const NodeDef& nodeDef, GenContext
         ShaderInput* input = getInput(nodeValue->getName());
         if (input)
         {
-            input->setPath(nodeValue->getNamePath());
+            string path = nodeValue->getNamePath();
```

Comment:
Fix shader reflection so that the interface input path will be returned instead of the downstream input. This removes any need for editors to try and find the upstream interface, but also makes it so that direct uniform will find the correct uniform in the uniform block.

Reply:
I can't leave a comment but the value evaluation at line 349 get's fixe by fix to `getInterfaceInput()` so that the correct value is returned.